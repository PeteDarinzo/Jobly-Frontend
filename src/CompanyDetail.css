
.CompanyDetail-title {
  position: relative;
  left: 15%;
  text-align: left;
  width: 65%;
  margin: 45px;
  background-color: white;
  color: dimgray;
  padding: 10px;
}

.CompanyDetail-list{
  width: 40%;
  margin: 50px auto 0 auto;
}

.CompanyDetail-title h1 {
  font-size: 2.5rem;
  color: darkblue;
}

.CompanyDetail-title h2 {
  font-size: 1.5rem;
}

.CompanyDetail h3 {
  margin: 20px;
} 