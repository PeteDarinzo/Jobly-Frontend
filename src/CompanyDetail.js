import React from "react";
import { useParams } from "react-router-dom";
import JobCard from "./JobCard";
import "./CompanyDetail.css";
import { v4 as uuid } from "uuid";

const CompanyDetail = ({ companies, jobs }) => {

  const { handle } = useParams();

  const company = companies.find(company => company.handle === handle);
  const openings = jobs.filter(job => job.companyName === company.name);
  console.log("openings: ", openings);

  return (
    <div>
      <div className="CompanyDetail-title">
        <h1>{company.name}</h1>
        <h2>{company.description}</h2>
      </div>
      <div>
        {openings.map(job => (
          <JobCard
            title={job.title}
            companyName={job.companyName}
            salary={job.salary}
            equity={job.equity}
            key={uuid()}
          />
        ))}
      </div>
    </div>
  );

}

export default CompanyDetail;