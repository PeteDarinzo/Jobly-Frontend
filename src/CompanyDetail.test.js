import CompanyDetail from "./CompanyDetail";
import { render, fireEvent } from '@testing-library/react';
import { MemoryRouter, Route } from "react-router-dom";



it("renders home with crashing", () => {
  render(
    <MemoryRouter initialEntries={["/companies/test"]}>
      <Route path="/companies/handle">
        <CompanyDetail companies={[]} jobs={[]} apply={() => { }} applications={() => { }} />
      </Route>
    </MemoryRouter>
  );
});

it("matches the home snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter initialEntries={["/companies/test"]}>
      <Route path="/companies/handle">
        <CompanyDetail companies={[]} jobs={[]} apply={() => { }} applications={() => { }} />
      </Route>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});