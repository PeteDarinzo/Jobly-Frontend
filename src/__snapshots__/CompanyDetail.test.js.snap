// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the home snapshot 1`] = `<DocumentFragment />`;
